### PR TITLE
Replace maven inter-module deps with copying src

### PR DESCRIPTION
**Pull request by relud**
This allows running maven in module directories and removes the need for running maven with `-P {module}` and `-DfailIfNoTests=false`, while avoiding duplication of maven plugin configuration, shared code, and shared dependencies.

### CODE REVIEW

File: ingestion-beam/checkstyle [hunk 1]
```diff
@@ -0,0 +1 @@
+../checkstyle/
```
Comment (relud) [inline ingestion-beam/checkstyle]: softlinks are necessary for checkstyle to work in both submodule directories and the parent module directory.

File: docs/ingestion-beam/sink-job.md [hunk 2]
```diff
@@ -174,14 +174,13 @@ Note: `-Dexec.args` does not handle newlines gracefully, but bash will remove
...
-`ingestion-beam/` directory; be aware, though, that Java 8 is the target JVM
+If you install Java and maven, you can invoke `mvn` in the following commands
+instead of using `./bin/mvn`; be aware, though, that Java 8 is the target JVM
 and some reflection warnings may be thrown on newer versions, though these are
 generally harmless.
 
-The provided `bin/mvn` script downloads and runs maven with the correct profile
-via docker so that less setup is needed on the local machine.
+The provided `bin/mvn` script downloads and runs maven via docker so that less
+setup is needed on the local machine.
```
Comment (jklukas) [inline docs/ingestion-beam/sink-job.md]: Can we go ahead and add a note here saying that `bin/mvn` is intended for getting started quickly and that for prolonged development, performance will be much better if you install the tools locally?

Comment (relud) [reply docs/ingestion-beam/sink-job.md]: added ```For prolonged development performance is
likely to be significantly better, especially in osx, if `mvn` is installed and
run natively without docker.```

File: docs/ingestion-beam/sink-job.md [hunk 3]
```diff
@@ -174,14 +174,15 @@ Note: `-Dexec.args` does not handle newlines gracefully, but bash will remove
...
+If you install Java and maven, you can invoke `mvn` in the following commands
+instead of using `./bin/mvn`; be aware, though, that Java 8 is the target JVM
 and some reflection warnings may be thrown on newer versions, though these are
 generally harmless.
 
-The provided `bin/mvn` script downloads and runs maven with the correct profile
-via docker so that less setup is needed on the local machine.
+The provided `bin/mvn` script downloads and runs maven via docker so that less
+setup is needed on the local machine. For prolonged development performance is
+likely to be significantly better, especially in osx, if `mvn` is installed and
```
Comment (jklukas) [inline docs/ingestion-beam/sink-job.md]:
```suggestion
likely to be significantly better, especially in MacOS, if `mvn` is installed and
```

pendantry

Comment (relud) [reply docs/ingestion-beam/sink-job.md]: wfm

File: pom.xml [hunk 4]
```diff
@@ -235,6 +244,66 @@
                     </execution>
                 </executions>
             </plugin>
+            <plugin>
+                <groupId>org.codehaus.mojo</groupId>
+                <artifactId>build-helper-maven-plugin</artifactId>
+                <version>3.0.0</version>
+                <executions>
+                    <!-- add ingestion-core/src to all modules -->
```
Comment (jklukas) [inline pom.xml]: I would not have thought of generate-sources as a method for making this work. This is cool!

**Pull request merged**